### PR TITLE
Simplify and document dispatch action strategy

### PR DESCRIPTION
**Pull request by msarahan**
This PR adds docs for the "dispatch action" pattern that we use for the telemetry actions. It also refactors the actions for simplicity, including changing out the go-based otel-cli for direct use of the official opentelemetry Python SDK.

### CODE REVIEW

**Comment by msarahan** [comment]
The change from otel-cli to the opentelemety python SDK isn't going as well as I thought/hoped. I'm going to split that off from this PR.

File: telemetry-impls/summarize/action.yml [hunk 1]
```diff
@@ -1,149 +1,37 @@
...
+      with:
+        all_jobs: true
+    - name: Upload job JSON file if debugging
+      uses: actions/upload-artifact@v4
+      if: runner.debug == '1'
+      with:
+        name: telemetry-tools-all_jobs.json
+        path: all_jobs.json
+    # This downloads ALL of the files that we have collected from each job.
+    - uses: actions/download-artifact@v4
```
Comment (ajschmidt8) [inline telemetry-impls/summarize/action.yml:29]: Given that we intend on eventually switching to using GitHub artifacts (see https://github.com/rapidsai/ops/issues/2982), I think we should create a prefix specifically for telemetry files and then use this action's `pattern` input to ensure that we don't needlessly download unrelated artifacts.

See here for more details: https://github.com/actions/download-artifact?tab=readme-ov-file#download-multiple-filtered-artifacts-to-the-same-directory

File: telemetry-impls/clean-up-artifacts/action.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+          artifacts.forEach(artifact => {
+            if (artifact.name.startsWith('telemetry-tools-')) {
+              github.rest.actions.deleteArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: artifact.id
+              });
+            }
+          });
```
Comment (ajschmidt8) [inline telemetry-impls/clean-up-artifacts/action.yml:52]:
```suggestion
          await Promise.all(
            artifacts
            .filter(artifact => artifact.name.startsWith('telemetry-tools-'))
            .map(artifact => {
              github.rest.actions.deleteArtifact({
                owner: context.repo.owner,
                repo: context.repo.repo,
                artifact_id: artifact.id
              });
            })
          );
```

The `github.rest.actions.deleteArtifact` function returns a promise that needs to be awaited. This suggestion splits the logic into a `.filter` and `.map` chain which will effectively return an array of promises that can all be awaited with `await Promise.all()`.

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**